ts/client/services/rpc.ts

```diff
@@ -19,70 +15,54 @@ const tx = {
 };
 
 export class RPC {
-    server: FastifyInstance;
-
-    constructor(server: FastifyInstance) {
-        this.server = server;
-    }
+    constructor(
+        { l2Storage }: CoreAPI,
+        fastify: FastifyInstance,
+        transferPool?: ITransferPool
+    ) {
+        fastify.get<{ Params: { stateID: number } }>(
+            "/user/state/:stateID",
+            async function(request) {
+                const stateID = request.params.stateID;
+                const state = await l2Storage.state.get(stateID);
+                return state.toJSON();
+            }
+        );
+        fastify.post<{ Body: { bytes: string } }>(
+            "/tx",
+            tx,
+            async (request, reply) => {
+                // TODO There should be a better way we can reject
+                // accepting transactions when not a proposer.
+                // Also, do we want to accept transactions when we are not
+                // the active proposer?
```

Comment:
Thoughts?

Reply:
Good question. Haven't given this a good thought yet. But my first thoughts are

- A proposer should not accept tx when not active.
- Ideally, the proposer can reply "hey the current active proposer endpoint is https://theotherprovider.com". But since we are talking to wallets, `reply.status(409).send("not a proposer")` seems fine. The application dev should handle this gracefully.

Reply:
Removed TODO, added note to https://github.com/thehubbleproject/hubble-contracts/issues/408#issuecomment-870768409